src/fromager/sdist.py

```diff
@@ -75,95 +75,105 @@ def handle_requirement(
         f'{req.name}: {"*" * (len(why) + 1)} handling {req_type} requirement {req} {why}'
     )
 
+    new_req, constraint = ctx.constraints.get_new_requirement(req)
```

Comment:
```suggestion
    constrained_req, constraint = ctx.constraints.get_new_requirement(req)
```

`new_req` isn't particularly descriptive. Maybe `constrained_req`? Alternatively, you could save `req` as `original_req` and then replace its value with the constrained value. That has the appeal of changing fewer lines, and for cases where there is no constraint it's the same value anyway.

Same comment about the method name, "new" doesn't tell us how it's different. Something like `constrain_requirement(req)` or `get_constrained_requirement(req)` explains what sort of change is going to happen.

Reply:
This is so much better, yes!